assets/js/payment-method-extensions/payment-methods/stripe/payment-request/index.js

```diff
@@ -16,30 +16,47 @@ const ApplePayPreview = () => <img src={ applePayImage } alt="" />;
 const canPayStripePromise = loadStripe();
 const componentStripePromise = loadStripe();
 
+let apiInitialized = false,
+	canPay = false;
+
+// Initialise stripe API client and determine if payment method can be used
+// in current environment (e.g. geo + shopper has payment settings configured).
+function paymentRequestAvailable( currencyCode ) {
+	// If we've already initialised, return the cached results.
+	if ( apiInitialized ) {
+		return canPay;
+	}
+
+	return canPayStripePromise.then( ( stripe ) => {
+		if ( stripe === null ) {
+			return false;
```

Comment:
Hmm we should probably set the cache here and return (because if stripe isn't in the callback then there's something wrong and non-recoverable, so no sense in repeated calls to the canMakePayment callback triggering additional checks).

As in my original example:

```js
if ( stripe === null ) {
    isInitialized = true;
    return canPay;
}
```